Implements remaining suggested improvements

FixItDik opened the pull request:
Resolves #7 (was already resolved in the previous pull request but I forgot to mention it in the pull request comment so it remained open).

Resolves #10 by adding a Pause button which pauses the timer and hides the grid when clicked (it also becomes "Resume" button which when clicked resumes the timer and shows the grid again)

This also resolves some minor issues with the best times button as well as adding some little code to take the player's name from any of their previous high scores rather than just the level they are playing.  There is some minor styling improvements for the grid positions that have 4 to 8 mines adjacent to them (were grey, now red) and removes the tab characters from the high score alert as most browsers do not support tab characters in alerts..

FixItDik committed:
Added Pause feature

Added a Pause feature but also fixed Best Times button so it works
(button had been renamed but new name not reflected in code) and also
does not try to use ```tab``` characters n alert message (as they are
not supported in most browsers)

FixItDik committed:
Fixed var naming mistake

I left the old convention in place, fixed now

FixItDik committed:
Commit issue fixed

For some reason the last line I changed in the previous changes did not
get committed, instead it appeared to have been deleted, very odd.

FixItDik committed:
Fixed code formatting issues

Sorry, I will get in to the swing of the 4 spaces and braces for single
line ```if```s I promise

FixItDik commented:
PS You will note I have implemented the ```else``` code formatting improvements for the X and Y params but have not implemented your infinite loop code, instead just removed the 3rd ```else``` from the numMines check which ensures there is always 1 non-mine spot on the grid (thus preventing an infinite loop)

FixItDik committed:
Merge remote-tracking branch 'refs/remotes/michaelbutler/master'

michaelbutler commented on js/MineSweeper.js:
```diff
@@ -645,14 +686,23 @@ jQuery(function ($) {
                 var bestTime = localStorage.getItem('best_time_' + level);
 
-                if (!bestTime || parseInt(time, 10) < parseInt(bestTime, 10)) {
+                if (!bestTime  || parseInt(time, 10) < parseInt(bestTime, 10)) {
```
extra space here needs to be removed

michaelbutler commented on js/MineSweeper.js:
```diff
@@ -215,19 +217,28 @@ jQuery(function ($) {
                     input.prop('disabled', true);
                 }
-                $('.new-game').trigger('click');
+                $('#newGame').trigger('click');
```
please change back to using a css class instead of id, it makes everything more versatile and portable.

michaelbutler commented on js/MineSweeper.js:
```diff
@@ -478,6 +516,7 @@ jQuery(function ($) {
 
             $('#mine_flag_display').val(numMines);
             $('#timer').val(0);
+            $('#pause').hide();
```
please make pause referenced by css class, not id. Ideally we want no id elements referenced anywhere in the code base. The reason for this is because you can only have 1 element of the same ID in the DOM at the same time.

FixItDik replied:
Just so I understand, when is it right to use an ID and when should we uses classes to identify elements we have created?  I personally would use a class when I have more than one element I wish to address but an ID when I have only the one, so in this case there is only one Pause button hence I chose the ID (much as the existing code uses '#level' for the level selector).  Just trying to understand, happy to change it :)

michaelbutler commented:
Sorry for the long delay, but I have some feedback. Thanks.

FixItDik commented:
Hi Michael, no worries mate, it's not like this is a mission critical project or anything.  I have filled my spare time with writing a JavaScript version of Spider Solitaire using some of the tricks I learnt from Minesweeper - still needs some tweaking but you're welcome to have a look (https://fixitdik.github.io/SpiderSolitaire/).  Mean time I will apply the fixes for your comments as soon as I get a chance.  Hope this finds you fit and well, Regards, Dik.

FixItDik committed:
Replaced all ID references with class references

The code no longer references any element by its ID but instead uses
class names.  Where I have added a class to enable this I have prefixed
the name with "ms" to represent MineSweeper in an attempt to avoid
confusion with any classes that may be in use on the page in which this
object is embedded.

FixItDik committed:
Fixed lines which became too long in last commit

I didn't realise there was a line length limit - apparently somewhere
between 110 and 117 characters - so have had to split some of the
constant strings near the end of the file making them a little untidy
but hey.

FixItDik commented:
Had some spare time over coffee :)

michaelbutler commented:
Merged in changes into https://github.com/michaelbutler/minesweeper/pull/18
